sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -1302,16 +1302,8 @@ public void setUrl(String url) {
     }
   }
 
-  @Test
-  public void testEnum() {
-    List<EnumBean> data = Arrays.asList(EnumBean.B);
-    Encoder<EnumBean> encoder = Encoders.bean(EnumBean.class);
```

Comment:
java enum is not java bean, I don't think it's a valid case. If we wanna support top level java enum, please add new API in `org.apache.spark.sql.Encoders`